src/zenml/zen_stores/schemas/trigger_schemas.py

```diff
@@ -0,0 +1,157 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""SQL Model Implementations for Triggers."""
+import base64
+import json
+from datetime import datetime
+from typing import Optional
+from uuid import UUID
+
+from sqlalchemy import TEXT, Column
+from sqlmodel import Field, Relationship
+
+from zenml.models.v2.core.trigger import (
+    TriggerRequest,
+    TriggerResponse,
+    TriggerResponseBody,
+    TriggerResponseMetadata,
+    TriggerUpdate,
+)
+from zenml.zen_stores.schemas.base_schemas import NamedSchema
+from zenml.zen_stores.schemas.schema_utils import build_foreign_key_field
+from zenml.zen_stores.schemas.user_schemas import UserSchema
+from zenml.zen_stores.schemas.workspace_schemas import WorkspaceSchema
+
+
+class TriggerSchema(NamedSchema, table=True):
+    """SQL Model for triggers."""
+
+    __tablename__ = "trigger"
+
+    workspace_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target=WorkspaceSchema.__tablename__,
+        source_column="workspace_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+    workspace: "WorkspaceSchema" = Relationship(back_populates="triggers")
+
+    user_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=UserSchema.__tablename__,
+        source_column="user_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+    user: Optional["UserSchema"] = Relationship(back_populates="triggers")
+
+    event_filter: bytes
```

Comment:
@stefannica by having this not as a relationship, but as an encoded json-byte - how would you go about filtering triggers by their event filter? Should we use hashes? This would mean for migrations we need to take out the even_filters - load them into the approriate classes and then store a new hash